Source/MQTTnet/Client/Connecting/MqttClientAuthenticateResult.cs

```diff
@@ -25,6 +26,8 @@ public class MqttClientAuthenticateResult
         public string ReasonString { get; set; }
 
         public ushort? ReceiveMaximum { get; set; }
+        
+        public MqttQualityOfServiceLevel? MaximumQoS { get; set; }
```

Comment:
In my opinion this should be not nullable. If the value is not present this should default to QoS 2 (From spec: Followed by a Byte with a value of either 0 or 1. It is a Protocol Error to include Maximum QoS more than once, or to have a value other than 0 or 1. If the Maximum QoS is absent, the Client uses a Maximum QoS of 2.). 

But not in this class. The conversion should be made in the data converter instead.